src/components/map/HfpMarkerLayer.js

```diff
@@ -72,9 +72,9 @@ class HfpMarkerLayer extends Component {
     runInAction(() => (this.positions = indexed));
   };
 
-  componentDidMount() {
+  async componentDidMount() {
     const {state, positions} = this.props;
-    this.indexPositions(positions);
+    await this.indexPositions(positions);
```

Comment:
Vehicle markers will not be drawn before indexing is done. Awaiting indexing fixes the bug where the vehicle markers weren't drawn immediately.